src/notes/io.js

```diff
@@ -0,0 +1,200 @@
+const concat = require('concat');
+
+const config = require('../config.js')();
+const log = require('../log.js');
+
+/**
+ * A stateless collection of functions for reading from and writing to files on disk
+ */
+module.exports = {
+    /**
+     * Load and initialise release extension config
+     * @returns {Object} data
+     */
+    async loadConfig() {
+        log.doing('Loading config');
+        let data = Object.assign({}, await config.load());
+
+        // Request github token if necessary
+        if (!data.token) {
+            setTimeout(() => opn('https://github.com/settings/tokens'), 2000);
+
+            const { inputToken } = await inquirer.prompt({
+                type: 'input',
+                name: 'inputToken',
+                message: 'I need a Github token, with "repo" rights (check your browser) : ',
+                validate: token => /[a-z0-9]{32,48}/i.test(token),
+                filter: token => token.trim()
+            });
+
+            data.token = inputToken;
+
+            await config.write(data);
+        }
+        return data;
+    },
+
+    /**
+     * Get 2 composer files from the user
+     * @returns {Object}
+     */
+    async getComposers() {
+        const file1 = await this.promptForComposerJson('Path to the starting composer.json');
+        const file2 = await this.promptForComposerJson('Path to the ending composer.json');
+        return await this.readComposerFiles({ file1, file2 });
+    },
+
+    /**
+     * Given 2 composer files, check them and convert the dependencies to JS objects
+     * @param {Object} files - should contain file1 & file2 paths
+     * @returns {Object}
+     */
+    async readComposerFiles(files) {
+        const { file1, file2 } = files;
+        const exts1 = await this.readComposerJsonDeps(file1);
+        const exts2 = await this.readComposerJsonDeps(file2);
+        return { exts1, exts2 };
+    },
+
+    /**
+     * Prompt the user for a path to a composer.json
+     * @param {String} [message='Path?']
+     * @param {String} [defaultFile='composer.json']
+     * @returns {String} path
+     */
+    async promptForComposerJson(message = 'Path?', defaultFile = 'composer.json') {
+        const { composerJsonPath } = await inquirer.prompt({
+            type: 'input',
+            name: 'composerJsonPath',
+            message,
+            default: `${process.cwd()}/${defaultFile}`
+        });
+
+        return composerJsonPath;
+    },
+
+    /**
+     * Read dependencies section from a given composer file on disk
+     * @param {String} composerJsonPath
+     * @returns {Object} composer dependencies
+     */
+    async readComposerJsonDeps(composerJsonPath) {
+        try {
+            const fileData = await fs.readJson(composerJsonPath, 'utf-8');
+            return fileData.require;
+        }
+        catch (err) {
+            log.error(err);
+            log.exit('Error occured while reading file.');
+        }
```

Comment:
Could be. But since this part of the program can be reached from 2 different scenarios (composers-as-params and composers-when-prompted), it's non-trivial to refactor this exit to do something meaningfully different (like a "try again | abort" prompt for example).

Therefore I suggest to change it only if a different behaviour is needed and strategised.